app/src/main/java/com/codepath/apps/critterfinder/activities/PetSearchFilterActivity.java

```diff
@@ -66,10 +75,13 @@ protected void onCreate(Bundle savedInstanceState) {
         // add a back button
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
         mSearchFilter = Parcels.unwrap(getIntent().getParcelableExtra(PetSearchFilterActivity.EXTRA_SEARCH_FILTER));
+        mPetService = new PetSearch(null);
+        mBreeds = new HashMap<SearchFilter.Species, List<Breed>>();
 
```

Comment:
@scottrichards - you mentioned this in skype and I think you are right. PetSearch should probably be a singleton with a
```
public static PetSearch getInstance() {
        return mInstance;
    }
```

and it would be helpful for me if the ctor didn't require a PetSearchCallbackInterface - we just pass in PetSearchCallbackInterface as an argument to the doPetSearch method. In my case from the search activity I'm searching for breeds and not a pet so I'm not implementing PetSearchCallbackInterface.

Happy to make that change but didn't want to cause merge conflicts on you.